internal/agent/agent_test.go

```diff
@@ -1,87 +1,35 @@
-package agent
+package agent_test
 
 import (
-	"context"
 	"os"
 	"path/filepath"
 	"testing"
 	"time"
 
 	"github.com/flightctl/flightctl/api/v1alpha1"
-	"github.com/flightctl/flightctl/internal/config"
 	"github.com/flightctl/flightctl/internal/util"
-	"github.com/flightctl/flightctl/pkg/log"
-	testutil "github.com/flightctl/flightctl/test/util"
+	"github.com/flightctl/flightctl/test/harness"
+
 	"github.com/stretchr/testify/require"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"sigs.k8s.io/yaml"
 )
 
 func TestDeviceAgent(t *testing.T) {
-	require := require.New(t)
-
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	testDirPath := t.TempDir()
-	defer t.Cleanup(func() {
```

Comment:
Oh this is missing, fixing.

Reply:
I will be leaving this outside of the test harness because ginkgo provides a TempDir function that provides automatic cleanup at the end of tests.